modules/json/src/main/java/com/spotify/ffwd/json/JsonObjectMapperDecoder.java

```diff
@@ -105,27 +105,24 @@ private Object decode0(ByteBuf in, List<Object> out) throws IOException {
     private Object decodeMetric(JsonNode tree, List<Object> out) {
         final String key = decodeString(tree, "key");
         final double value = decodeDouble(tree, "value");
-        final Date time = decodeTime(tree, "time");
+        final long time = decodeTime(tree, "time");
         final Optional<String> host = Optional.ofNullable(decodeString(tree, HOST));
-        final Set<String> riemannTags = decodeTags(tree, "tags");
         final Map<String, String> tags = decodeAttributes(tree, "attributes");
         final Map<String, String> resource = decodeResources(tree, "resource");
-        final String proc = decodeString(tree, "proc");
 
         host.ifPresent(h -> tags.put(HOST, h));
 
-        return new Metric(key, value, time, riemannTags, tags, resource, proc);
+        return new Metric(key, Value.DoubleValue.create(value), time, tags, resource);
     }
 
-    private Date decodeTime(JsonNode tree, String name) {
+    private long decodeTime(JsonNode tree, String name) {
         final JsonNode n = tree.get(name);
 
         if (n == null) {
-            return null;
+            return 0;
```

Comment:
Did you test that this would create metric with current timestamp? I believe the old code by setting it to null insures it will be set to current time downstream.